### PR TITLE
Fix Sentence Construction

### LINKED ISSUE #23021: Rewrite sentence in "Allowing changes to a pull request branch created from a fork"
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

### What part(s) of the article would you like to see updated?

This sentence is hard to understand. It's too long and tries to explain three things at once:

> Only pull request authors can give upstream repository maintainers, or those with push access to the upstream repository, permission to make commits to their pull request's compare branch in a user-owned fork.

This _one_ sentence mentions:

- Only pull request authors can give permission
- Lists who can commit: upstream maintainers / those with push access to upstream
- That the branch needs to be on a user-owned fork

This sentence needs a rewrite.

### Additional information

Trying one more time to get this past your @docubot:

- #23020

### PR DESCRIPTION
**Pull request by Sriparno08**
### Why:

The below sentence is hard to understand:

> Only pull request authors can give upstream repository maintainers, or those with push access to the upstream repository, permission to make commits to their pull request's compare branch in a user-owned fork.

Hence, I have decided to change the sentence construction to make it a bit easier to understand.

Closes #23021

### What's being changed (if available, include any code snippets, screenshots, or gifs):

I have changed the sentence to:

Screenshot 2023-01-08 115546

### Check off the following:

- [x] I have reviewed my changes in staging (look for the "Automatically generated comment" and click the links in the "Preview" column to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

**Comment by cmwilson21** [comment]
@Sriparno08 Thanks for the PR! The original issue is still waiting for review, so once that happens, someone will review your PR based on the issue's review. ✨ 

In the mean time, please take a look at our [help wanted](https://github.com/github/docs/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22) section to find open issues you can work on. All items on this board have already been approved and have directions or a plan to get contributors started. 👍

**Comment by Sriparno08** [comment]
Alright, @cmwilson21 👍

**Comment by isaacmbrown** [comment]
Hi @Sriparno08, thanks for the PR ✨ I think your proposal will clear up a lot of confusion with a small adjustment:

Although it wasn't very clear from the original wording, in this context `upstream repository maintainer` _means_ anyone with push access to the upstream repository. Your proposal makes it sound like you can choose to give access _either_ only to people with the maintainer role, _or_ to anyone with push access to the upstream. If you look at the instructions on the page, you'll see that this level of granularity isn't available:

> On user-owned forks, if you want to allow anyone with push access to the upstream repository to make changes to your pull request, select **Allow edits from maintainers**.

If you could edit your wording to make that clearer, I think this will be looking great! 🚀

**Comment by Ahujco1986** [comment]
Unknown

**Comment by Sriparno08** [comment]
Okay @isaacmbrown! Let me fix it!

**Comment by Sriparno08** [comment]
Hey @isaacmbrown, I have made some changes. Have a look!

File: content/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork.md
Comment (isaacmbrown) [inline content/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork.md:19]:
```suggestion
When a user creates a pull request from a fork that they own, the user generally has the authority to decide if other users can commit to the pull request's compare branch. If the pull request author wants greater collaboration, they can grant maintainers of the upstream repository (that is, anyone with push access to the upstream repository) permission to commit to the pull request's compare branch. To learn more about upstream repositories, see "[About forks](/pull-requests/collaborating-with-pull-requests/working-with-forks/about-forks)."
```

**Comment by isaacmbrown** [comment]
Hi @Sriparno08, looks good, thanks for making that change ✨

I just made a couple of changes with our style guide and translation workflow in mind. Otherwise this look good to go, and I'll add a label indicating that this can be merged 🚢

**Review by isaacmbrown — APPROVED** [review]

**Comment by Sriparno08** [comment]
Thanks, @isaacmbrown 👍

**Pull request merged**
